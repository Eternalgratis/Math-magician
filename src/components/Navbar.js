import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const urls = [
    {
      id: 1,
      url: '/',
      text: 'Home',
    },
    {
      id: 2,
      url: '/Calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      url: '/Quote',
      text: 'Quote',
    },
  ];
  return (
    <nav className="nav-container">
      <h1 className="nav-header">Math Magicians</h1>
      <ul className="nav-list">
        {urls.map((url) => (
          <li key={url.id}>
            <NavLink to={url.url}>{ url.text}</NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
