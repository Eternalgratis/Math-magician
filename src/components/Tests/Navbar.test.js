import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar tests', () => {
  test('Renders Navbar component', () => {
    const tree = Renderer.create(<BrowserRouter><Navbar /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
