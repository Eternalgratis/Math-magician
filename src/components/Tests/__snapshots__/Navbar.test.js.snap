// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests Renders Navbar component 1`] = `
<nav
  className="nav-container"
>
  <h1
    className="nav-header"
  >
    Math Magicians
  </h1>
  <ul
    className="nav-list"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
